### PR TITLE
Fix benchmark edge cases, tighten CLI validation, and expand unit test coverage

### PR DESCRIPTION
**Pull request by irvineoy**
## Summary

This PR fixes several correctness and robustness issues across the project, cleans up repo-wide lint warnings, and adds broader unit test coverage for core helper logic.

## What Changed

- Tightened kernel type validation:
  - `parse_kernel_type()` now raises an error for unsupported values
  - CLI and YAML-based config loading now fail fast instead of silently falling back to HIP

- Made multi-source compilation output naming deterministic:
  - replaced Python's process-randomized `hash()` usage with a stable hash-based naming scheme
  - applies to both HIP and CUDA multi-file compilation paths

- Cleaned up repo-wide lint issues reported by `ruff`:
  - removed unused imports
  - removed unnecessary f-strings
  - simplified a few small code paths for clarity

## Tests Added / Updated

- Added coverage for stricter kernel type validation
- Added coverage for deterministic compile output naming

## Why

These changes make the project safer and easier to debug by:
- preventing silent misconfiguration
- avoiding incorrect distributed benchmark behavior
- making compile outputs reproducible across processes

## Validation

Ran:

```bash
.venv/bin/python -m ruff check Magpie tests
.venv/bin/python -m pytest -q

### CODE REVIEW

**Comment by haofrank** [comment]
about ray have you tested it on ray cluster?

**Comment by haofrank** [comment]
Marking as `Draft` until #18 merges, this PR currently repeats those commits. After #18 is on main, please rebase onto main and flip back to Ready for review. Thanks!

File: Magpie/main.py [hunk 1]
```diff
@@ -66,7 +66,13 @@ def parse_kernel_type(type_str: str) -> KernelType:
         "py": KernelType.PYTORCH,
         "triton": KernelType.TRITON,
     }
-    return type_map.get(type_str.lower(), KernelType.HIP)
+    normalized = type_str.lower()
```
Comment (sinarafati-amd) [inline Magpie/main.py]: it seems the default fall back was hip for any unknown input; 

function now throws an exception. which makes sense but we need to make sure it does not break anything due to behavior change

File: Magpie/main.py [hunk 2]
```diff
@@ -484,9 +490,14 @@ def run_analyze(args, config: Dict[str, Any]) -> int:
...
-        kernel_configs, perf_overrides, corr_overrides, sched_overrides = (
-            load_kernel_config(args.kernel_config)
-        )
+        try:
+            kernel_configs, perf_overrides, corr_overrides, sched_overrides = (
+                load_kernel_config(args.kernel_config)
+            )
+        except ValueError as e:
+            logger.error(f"Invalid kernel config: {e}")
+            print(f"Error: {e}")
```
Comment (haofrank) [inline Magpie/main.py]: The `print` statement here is not useful to us; let's remove it.

File: Magpie/main.py [hunk 3]
```diff
@@ -497,7 +508,12 @@ def run_analyze(args, config: Dict[str, Any]) -> int:
             print("Error: --testcase is required for analyze mode")
             return 1
 
-        kernel_type = parse_kernel_type(args.type)
+        try:
+            kernel_type = parse_kernel_type(args.type)
+        except ValueError as e:
+            logger.error(str(e))
+            print(f"Error: {e}")
```
Comment (haofrank) [inline Magpie/main.py]: same useless `print`

File: Magpie/main.py [hunk 4]
```diff
@@ -612,11 +628,21 @@ def run_compare(args, config: Dict[str, Any]) -> int:
...
-        kernel_configs, perf_overrides, corr_overrides, sched_overrides = (
-            load_kernel_config(args.kernel_config)
-        )
+        try:
+            kernel_configs, perf_overrides, corr_overrides, sched_overrides = (
+                load_kernel_config(args.kernel_config)
+            )
+        except ValueError as e:
+            logger.error(f"Invalid kernel config: {e}")
+            print(f"Error: {e}")
```
Comment (haofrank) [inline Magpie/main.py]: same useless `print`

**Review by haofrank — APPROVED** [review]
LGTM now.
